src/main/java/com/amazon/ion/impl/bin/IonRawBinaryWriter_1_1.kt

```diff
@@ -114,50 +140,155 @@ class IonRawBinaryWriter_1_1 internal constructor(
         buffer.writeBytes(_Private_IonConstants.BINARY_VERSION_MARKER_1_1)
     }
 
+    /**
+     * Ensures that there is enough space in the annotation buffers for [n] annotations.
+     * If more space is needed, it over-allocates by 8 to ensure that we're not continually allocating when annotations
+     * are being added one by one.
+     */
+    private inline fun ensureAnnotationSpace(n: Int) {
+        if (annotationsIdBuffer.size < n || annotationsTextBuffer.size < n) {
```

Comment:
Should these grow independently? Does that significantly complicate things?

Reply:
The way I have it implemented right now, if position `x` of `annotatationsTextBuffer` is null, then we know to use the value from `annotationsIdBuffer`, so it does simplify the write logic if `annotatationsTextBuffer` is at least as large as `annotationsIdBuffer`, and if one has to be as least as large as the other, it simplifies the "growing" logic if we just make them the same size. I am not too concerned about the efficiency of growing both arrays at the same time because realistically, we don't expect there to be 10+ annotations for nearly all use cases.